From 702c48d2aeb5da3b92e2b01ddd33e49c843d4df2 Mon Sep 17 00:00:00 2001
From: "tao.dong" <tao.dong@amlogic.com>
Date: Mon, 23 Dec 2013 17:57:13 +0800
Subject: [PATCH 2646/5965] it cause resume fail in HDMI 4k2k output mode
 Revert "PD #84336: m8: power: Dynamic control suspend arc log & support
 transfer param to suspend fimware"

This reverts commit 6376e20d2a12806f2984fcc776589729dcc25b51.
---
 arch/arm/mach-meson8/power_suspend.c | 22 ----------------------
 kernel/sysctl.c                      | 12 ------------
 2 files changed, 34 deletions(-)
 mode change 100755 => 100644 kernel/sysctl.c

diff --git a/arch/arm/mach-meson8/power_suspend.c b/arch/arm/mach-meson8/power_suspend.c
index 274f23e23c2e..73fb3dbff9c0 100755
--- a/arch/arm/mach-meson8/power_suspend.c
+++ b/arch/arm/mach-meson8/power_suspend.c
@@ -50,18 +50,6 @@
 #define APPF_SAVE_DEBUG        (1<<3)
 #define APPF_SAVE_L2           (1<<4)
 
-/******************
-***You need sync this param struct with arc_pwr.h is suspend firmware.
-***1st word is used for arc output control: serial_disable.
-***2nd word...
-***
-***If you need transfer more params, you need sync the struck define in arc_pwr.h
-*******************/
-#define PARAM_ADDR             (IO_SRAM_BASE + 0x10200)
-
-unsigned int arc_serial_disable;
-
-
 #if 0
 #ifdef CONFIG_HARDWARE_WATCHDOG
 void disable_watchdog(void)
@@ -86,15 +74,6 @@ void reset_watchdog(void)
 #endif /* CONFIG_HARDWARE_WATCHDOG */
 #endif
 
-static void check_in_param(void)
-{
-	unsigned int p_addr;
-
-	p_addr = PARAM_ADDR;
-	*((unsigned int *)p_addr) = arc_serial_disable;
-
-	return;
-}
 
 int meson_power_suspend(void)
 {
@@ -103,7 +82,6 @@ int meson_power_suspend(void)
 	unsigned p_addr;
 	void	(*pwrtest_entry)(unsigned,unsigned,unsigned,unsigned);
 
-	check_in_param();
 	flush_cache_all();
 
 	addr = 0x04F04400;//entry.s start
diff --git a/kernel/sysctl.c b/kernel/sysctl.c
old mode 100755
new mode 100644
index c32311853aad..1a07798c9146
--- a/kernel/sysctl.c
+++ b/kernel/sysctl.c
@@ -273,9 +273,6 @@ static int min_extfrag_threshold;
 static int max_extfrag_threshold = 1000;
 #endif
 
-#ifdef CONFIG_MESON_SUSPEND
-extern unsigned int arc_serial_disable;
-#endif
 static struct ctl_table kern_table[] = {
 	{
 		.procname	= "sched_child_runs_first",
@@ -1066,15 +1063,6 @@ static struct ctl_table kern_table[] = {
 		.mode		= 0644,
 		.proc_handler	= proc_dointvec,
 	},
-#endif
-#ifdef CONFIG_MESON_SUSPEND
-	{
-		.procname	= "arc_firm_log",
-		.data		= &arc_serial_disable,
-		.maxlen		= sizeof(unsigned int),
-		.mode		= 0644,
-		.proc_handler	= proc_dointvec,
-	},
 #endif
 	{ }
 };
-- 
2.19.0

